compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -419,6 +419,13 @@ enum AnonymousLifetimeMode {
     PassThrough,
 }
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum ConditionScope {
```

Comment:
Why have separate LetSource and ConditionScope types?

Reply:
Yeah I guess they are interchangeable. It's just that `LetSource` makes more sense in `ExprKind::Let` and `Option<ConditionScope>` makes more sense in `LoweringContext`. Should I just use `LetSource`?